sycl/source/detail/scheduler/commands.cpp

```diff
@@ -239,8 +239,20 @@ void Command::waitForEvents(QueueImplPtr Queue,
   if (!EventImpls.empty()) {
     std::vector<RT::PiEvent> RawEvents = getPiEvents(EventImpls);
     if (Queue->is_host()) {
-      const detail::plugin &Plugin = EventImpls[0]->getPlugin();
-      Plugin.call<PiApiKind::piEventsWait>(RawEvents.size(), &RawEvents[0]);
+      // host queue can wait for events from different contexts
+      std::map<context_impl *, std::vector<EventImplPtr>>
+          RequiredEventsPerContext;
+
+      for (const EventImplPtr &Event : EventImpls) {
+        ContextImplPtr Context = Event->getContextImpl();
+        RequiredEventsPerContext[Context.get()].push_back(Event);
```

Comment:
Maybe you should assert `Context.get()` is not nullptr here? Can there be a dependency to an event on a host queue?

Reply:
> Can there be a dependency to an event on a host queue?

No, there can't be such dependency 'cause the events here are only `MPreparedDepsEvents` and host events are gathered only in `MPreparedHostDepsEvents`.

Reply:
Added assertions to prevent user's fault in future.